ar_read.c

```diff
@@ -113,12 +113,16 @@ static int ar_read(lua_State *L) {
     lua_getfield(L, 1, "reader"); // {ud}, {fenv}, fn
     if ( ! lua_isfunction(L, -1) ) err("MissingArgument: required parameter 'reader' must be a function");
     lua_setfield(L, -2, "reader"); // {ud}, {fenv}
-    lua_setfenv(L, -2); // {ud}
+#if LUA_VERSION_NUM > 501
```

Comment:
Can this be refactored into ar.h (the new header I mentioned) so it looks something like this instead?

```
#if LUA_VERSION_NUM <= 501
#    define lua_setuservalue lua_setfenv
#endif
```